File: core/execution_physics/semantic_runtime_orbit_engine.py
from __future__ import annotations

from typing import Any, Dict, List


MAX_ORBITS = 1000


def compute_runtime_orbits(
    runtime_ir: Dict[str, Any],
) -> Dict[str, Any]:
    topology = runtime_ir.get("distributed_topology", {})
    nodes = sorted(
        topology.get("nodes", []),
        key=lambda x: str(x.get("id")),
    )[:MAX_ORBITS]
    orbits = [
        {
            "orbit": idx,
            "node": node.get("id"),
        }
        for idx, node in enumerate(nodes)
    ]
    return {
        "orbits": orbits,
        "orbit_count": len(orbits),
        "bounded": True,
    }

File: src/execution_physics/semanticRuntimeOrbitEngine.ts
/**
 * Converted from Python: core/execution_physics/semantic_runtime_orbit_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export let MAX_ORBITS: any = 1000;
export function computeRuntimeOrbits(runtime_ir: any): any {
  var topology: any = py.get(runtime_ir, "distributed_topology", {});
  var nodes: any = py.slice(py.sorted(py.get(topology, "nodes", []), {key: ((x: any) => py.toStr(py.get(x, "id"))) as (item: any) => any}), null, MAX_ORBITS);
  var orbits: any = py.enumerate(nodes).map(([idx, node]: any) => ({"orbit": idx, "node": py.get(node, "id")}));
  return {"orbits": orbits, "orbit_count": py.len(orbits), "bounded": true};
}
